xarray/coding/cftime_offsets.py

```diff
@@ -41,18 +41,17 @@
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 import re
-import typing
 from datetime import timedelta
 from functools import partial
+from typing import TYPE_CHECKING
 
 import numpy as np
 
-from ..core.pycompat import TYPE_CHECKING
 from .cftimeindex import CFTimeIndex, _parse_iso8601_with_reso
 from .times import format_cftime_datetime
 
 if TYPE_CHECKING:
-    from typing import ClassVar, Optional
+    from typing import ClassVar  # Requires Python >= 3.5.3
```

Comment:
We can bump all the way to Python 3.5.3 if you prefer :)

Reply:
Dude, if it were for me I'd bump it to 3.7 😂
It's up for you to decide...

Reply:
Let's bump it up to the lowest 3.5.x version that has all the typing utilities we want.

Reply:
Bumped to 3.5.3 and ready to merge